backend/apps/volontulo/views/api.py

```diff
@@ -65,6 +69,62 @@ def login_view(request):
     )
 
 
+@api_view(['POST'])
+@authentication_classes((CsrfExemptSessionAuthentication,))
+@permission_classes((AllowAny,))
+def register_view(request):
+    """REST API register view."""
+    if not request.user.is_authenticated():
+        email = request.data.get('email')
+        password = request.data.get('password')
+
+        try:
+            user = User.objects.create_user(
+                username=email,
+                email=email,
+                password=password,
+                is_active=False
+            )
+
+            user.save()
+            profile = UserProfile(user=user)
+            ctx = {'token': profile.uuid, 'angular_root':
+                   settings.ANGULAR_ROOT}
+            profile.save()
+            send_mail(request, 'registration', [user.email], context=ctx)
+        except IntegrityError:
+            pass
+
+        return Response(
+            status=status.HTTP_201_CREATED,
+        )
+
+    return Response(
+        serializers.UserSerializer(request.user, context={
+            'request': request
+        }).data,
+        status=status.HTTP_400_BAD_REQUEST,
+    )
+
+
+@api_view(['POST'])
+@authentication_classes((CsrfExemptSessionAuthentication,))
+@permission_classes((AllowAny,))
+def activate(request):
+    """View responsible for activating user account."""
+    try:
+        profile = UserProfile.objects.get(uuid=request.data.get('uuid'))
+        profile.user.is_active = True
+        profile.user.save()
+        return Response(
+            status=status.HTTP_201_CREATED,
+        )
+    except UserProfile.DoesNotExist:
```

Comment:
this exception can only occur on the line 116, not the following lines. Please catch only on this one.

Reply:
+1